rs/replicated_state/src/metadata_state.rs

```diff
@@ -1569,199 +1553,30 @@ impl Streams {
-    pub fn get_mut_or_insert(&mut self, destination: SubnetId) -> StreamHandle {
-        // Can't (easily) validate stats when `StreamHandle` gets dropped, but we should
+    pub fn get_mut_or_insert(&mut self, destination: SubnetId) -> &mut Stream {
+        // Can't (easily) validate stats when `Stream` gets dropped, but we should
         // at least do it before.
         #[cfg(debug_assertions)]
         self.debug_validate_stats();
 
-        StreamHandle::new(
-            self.streams.entry(destination).or_default(),
-            &mut self.guaranteed_responses_size_bytes,
-        )
-    }
-
-    /// Returns the guaranteed response sizes by responder canister stat.
-    pub fn guaranteed_responses_size_bytes(&self) -> &BTreeMap<CanisterId, usize> {
-        &self.guaranteed_responses_size_bytes
-    }
-
-    /// Prunes zero-valued guaranteed response sizes entries.
-    ///
-    /// This is triggered explicitly by `ReplicatedState` after it has updated the
-    /// canisters' copies of these values (including the zeroes).
-    pub fn prune_zero_guaranteed_responses_size_bytes(&mut self) {
-        self.guaranteed_responses_size_bytes
-            .retain(|_, &mut value| value != 0);
-    }
-
-    /// Computes the `guaranteed_responses_size_bytes` map from scratch. Used when
-    /// deserializing and in asserts.
-    ///
-    /// Time complexity: O(num_messages).
-    pub fn calculate_stats(streams: &StreamMap) -> BTreeMap<CanisterId, usize> {
-        let mut guaranteed_responses_size_bytes: BTreeMap<CanisterId, usize> = BTreeMap::new();
-        for (_, stream) in streams.iter() {
-            for (_, msg) in stream.messages().iter() {
-                if let RequestOrResponse::Response(response) = msg {
-                    if response.deadline == NO_DEADLINE {
-                        *guaranteed_responses_size_bytes
-                            .entry(response.respondent)
-                            .or_default() += msg.count_bytes();
-                    }
-                }
-            }
-        }
-        guaranteed_responses_size_bytes
+        self.streams.entry(destination).or_default()
     }
 
     /// Checks that the running accounting of the sizes of responses in streams is
     /// accurate.
     #[cfg(debug_assertions)]
-    fn debug_validate_stats(&self) {
-        let mut nonzero_guaranteed_responses_size_bytes =
-            self.guaranteed_responses_size_bytes.clone();
-        nonzero_guaranteed_responses_size_bytes.retain(|_, &mut value| value != 0);
-        debug_assert_eq!(
-            Streams::calculate_stats(&self.streams),
-            nonzero_guaranteed_responses_size_bytes
-        );
-    }
-}
-
-/// A mutable reference to a stream owned by a `Streams` struct; bundled with
-/// the `Streams`' stats, to be updated on stream mutations.
-pub struct StreamHandle<'a> {
-    stream: &'a mut Stream,
-
-    guaranteed_responses_size_bytes: &'a mut BTreeMap<CanisterId, usize>,
-}
-
-impl<'a> StreamHandle<'a> {
```

Comment:
Was `guaranteed_response_size_bytes` really the only reason for the existence of `StreamHandle`? I seem to remember there were also some encapsulation related reasons.